app/controllers/GrpcServerImpl.java

```diff
@@ -0,0 +1,12 @@
+package controllers;
+
+import java.lang.annotation.*;
+import javax.inject.Qualifier;
+import static java.lang.annotation.ElementType.*;
+
+@Target({METHOD, FIELD, PARAMETER, TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Qualifier
+public @interface GrpcServerImpl {
```

Comment:
This is most interesting, IMO. I will research a little to check if Guice can somehow automatically discover/apply these annotations (although I usually don't like too much magic).

Reply:
It seemed promising, but did not appear to work in Guice (taking mostly the same code path as for `Named`)

if Guice indeed never does classpath scanning as @johanandren mentioned it makes sense that this would not work ;)